Lib/posixpath.py

```diff
@@ -402,7 +402,9 @@ def realpath(filename, *, strict=False):
         curdir = '.'
         pardir = '..'
         getcwd = os.getcwd
-    return _realpath(filename, strict, sep, curdir, pardir, getcwd)
+    maxlinks = 40 if strict else None  # TODO: use limit set by OS
```

Comment:
I'd use a global constant for the default. That way you can: either use the OS constant, or set it yourself (and it will be easier for your tests I think).

Reply:
I'm not sure how the OS limit will be retrieved, better to leave it hard coded for now.
I can add a comment to keep the limit in sync with the test though.

Reply:
It does not really matter how we retrieve it or not later I think. But a global constant could make it customizable for users as well (we use those kind of patterns, e.g., in the recent `multiprocessing.connection.Connection` where the pagesize was hardcoded as a global constant).

Reply:
Isn't it a problem that it's only customisable on Unix? Otherwise I would have probably exposed this as a parameter.

Reply:
For the pagesize? not really. But we also have for instance BUFFER_SIZE constants in pickle & co modules.

Reply:
I meant for MAXLINKS, on Windows we use the native functions.

Reply:
Oh, well let's make the constant private then. It will make tests easier to maintain IMO. We can just put any constant we want (the 40 is arbitrary so if it's arbitrary, making it a private constant is better than just putting comments everywhere and less prone to errors in the future).

Reply:
Or you can just make a private function that returns 40 and then we'll see how to make it OS-dependent?